includes/admin/home/tasks/class-sensei-home-tasks-provider.php

```diff
@@ -148,4 +155,57 @@ private function get_course() {
 	public function mark_as_completed( $completed = true ) {
 		update_option( self::COMPLETED_TASKS_OPTION_KEY, $completed, false );
 	}
+
+	/**
+	 * Tells if the WP hooks were attached or not.
+	 *
+	 * @var bool
+	 */
+	private static $attached_hooks = false;
+
+	/**
+	 * The option name that wp-calypso automatically fetches to use
+	 * as a reference for Launchpad tasks' complete statuses.
+	 *
+	 * @var string
+	 */
+	const CALYPSO_LAUNCHPAD_STATUSES_NAME = 'launchpad_checklist_tasks_statuses';
+
+	/**
+	 * Attaches required hooks.
+	 */
+	private function attach_tasks_statuses_hooks() {
+		// Attach the hooks only once.
+		if ( self::$attached_hooks ) {
+			return;
+		}
+		self::$attached_hooks = true;
+
+		// Attach the hooks only on atomic sites.
+		if ( ! Sensei_Utils::is_atomic_platform() ) {
+			return;
+		}
+
+		// Attach the update_tasks_statuses method to filters and actions
+		// that can affect the status of the Sensei Home tasks.
+		add_filter( 'save_post_course', [ $this, 'update_tasks_statuses' ] );
+		add_action( 'wp_ajax_sensei_settings_section_visited', [ $this, 'update_tasks_statuses' ] );
```

Comment:
I am probably missing a discussion here but it seems that this step is name 'Publish your first course' but we complete it if the user simply saved a course or visited the settings.

Also there is only one task although in the designs it looks like it should be many ('Create your first course', 'Configure learning mode' etc). Did we decide to drop these?

Reply:
> I am probably missing a discussion here but it seems that this step is name 'Publish your first course' but we complete it if the user simply saved a course or visited the settings.

We do not complete any task by ourselves, we simply update the `sensei_home_tasks_list_is_completed` options when these events/hooks are fired. Each time the hook is fired it checks gets the statuses of those tasks from respective Task instances. If, at that point it turns out the user didn't publish any courses the task will be marked as incomplete.

The same goes for all other tasks as well. The changes in this PR do not contain any logic to mark a task to complete/incomplete. We are simply hooking into events that might affect the statuses of the tasks and make sure it is reflected in `sensei_home_tasks_list_is_completed` wp option. So that when calypso fetches this option it is up to date.

> Also there is only one task although in the designs it looks like it should be many ('Create your first course', 'Configure learning mode' etc). Did we decide to drop these?

The tasks in the designs are simply a copy of the Sensei Home tasks. I don't think tasks from the Sensei Home page makes a lot of sense for Sensei Launchpad. The tasks in the Launchpad should be things that are required to complete in order to launch your course site. The tasks on the Sensei Home page is more about completing all the setups that Sensei offers in order to take full advantage of it. At least that's how I understand it.

On the team call on November 10th, when we discussed about the tasks on the Sensei Launchpad and Sensei Home page. Ronnie agreed to drop off the "Configure Learning Mode" and "Sell your Course with WooCommerce" tasks and leave on the "Publish your first Course" task in order to get the user to the finish like quicker. But I might remember it wrong. CC @burtrw to confirm this.

That being said. The changes in this pr makes sure all the tasks from the Sensei Home page are set into the `sensei_home_tasks_list_is_completed` wp options so it is available for Sensei Launchpad in calypso. So even if we decide we want to add "Confirgure Learning Mode" and other tasks from Sensei Home page to Sensei Launchpad we won't need any changes in this PR.

Reply:
> The changes in this PR do not contain any logic to mark a task to complete/incomplete

Ah somehow I missed that. Thank you for the explanation, these all make sense.